packages/thirdweb/src/react/web/ui/prebuilt/NFTMedia/index.tsx

```diff
@@ -0,0 +1,122 @@
+"use client";
+
+import type { Chain } from "../../../../../chains/types.js";
+import { MediaRenderer } from "../../MediaRenderer/MediaRenderer.js";
+import type { MediaRendererProps } from "../../MediaRenderer/types.js";
+import { useNFTProviderContext } from "../NFTProvider/index.js";
+import { useNFTMedia } from "../NFTProvider/useNFTMedia.js";
+
+export type NFTMediaProps = MediaRendererProps & {
+  // The NFT contract address
+  contractAddress?: string;
+  // The chain which the NFT contract was deployed on
+  chain?: Chain;
+  // The tokenId whose media you want to load
+  tokenId?: bigint;
+
+  /**
+   * By standard format we look for the `image` field from the metadata (e.g: `const image = metadata.image`)
+   * However, some contracts do not follow the standard. In that case users can use this field to override the default
+   */
+  overrideMediaField?: string;
```

Comment:
seems very niche?

Reply:
Got cha!

Reply:
the ones we care about are:
`animation_url`
`image`
`image_url`
`external_url`

in that order

if `animation_url` is set **and** `image` or `image_url` are set then `animation_url` should be the primary source and `image` or `image_url` should be used as a "poster image"